File: 30DaysOfCode/solution_14.py
class Difference:
    def __init__(self, a):
        self.__elements = a
        self.__elements_size = len(self.__elements)
        self.maximumDifference = None

    def computeDifference(self):
        maximumDifference = float()
        for i_idx, __ in enumerate(self.__elements):
            for j_idx, ___ in enumerate(self.__elements):
                current_difference = self.__elements[i_idx] - self.__elements[j_idx]

                if i_idx == 0 and j_idx == 0:
                    maximumDifference = current_difference
                elif current_difference > maximumDifference:
                    maximumDifference = current_difference
        self.maximumDifference = maximumDifference


# End of Difference class

_ = input()
a = [int(e) for e in input().split(" ")]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)

File: 14-Scope/solution_14.scala
// Calculate & display the max absolute difference of an immutable sequence.
// Task requires using a Difference class, that applies a 'computeDifference' method. 

import scala.io.StdIn.readLine

class Difference(a: Seq[Int]) {
    private val elements: Seq[Int] = a
    private val elementsSize: Int = elements.length
    var maximumDifference: Option[Int] = None

    def computeDifference(): Unit = {
        var maxDiff = 0
            for (i <- 0 until elementsSize) {
                for (j <- 0 until elementsSize) {
                      val currentDifference = scala.math.abs(elements(i) - elements(j))
                      if (currentDifference > maxDiff) {
                          maxDiff = currentDifference
                      }
                  }
              }
          maximumDifference = Some(maxDiff)
      }
}

object Solution {
    def main(args: Array[String]): Unit = {
        readLine() // discard the first input line as in py
        val a = readLine().split(" ").map(_.toInt)

        val d = new Difference(a)
        d.computeDifference()

        println(d.maximumDifference.get)
    }
}
